MELA/interface/MadMela.h

```diff
@@ -0,0 +1,70 @@
+            mdl_sth__exp__4,mdl_sth__exp__6,mdl_sth__exp__5,mdl_dwz,mdl_g1sh,mdl_gwsh,mdl_dmw,mdl_dwt,mdl_dww,mdl_dwhc,mdl_dwhb,mdl_dwhta,mdl_vevhat__exp__3,mdl_cth__exp__4,mdl_cth__exp__5,mdl_ckmlambda,mdl_ckma,mdl_ckmrho,mdl_ckmeta,
+            mdl_cg,mdl_cw,mdl_ch,mdl_chbox,mdl_chdd,mdl_chg,mdl_chw,mdl_chb,mdl_chwb,mdl_cehre,mdl_cuhre,mdl_cdhre,mdl_cewre,mdl_cebre,mdl_cugre,mdl_cuwre,mdl_cubre,mdl_cdgre,mdl_cdwre,mdl_cdbre,mdl_chl1,mdl_chl3,mdl_che,mdl_chq1,
+            mdl_chq3,mdl_chu,mdl_chd,mdl_chudre,mdl_cll,mdl_cll1,mdl_cqq1,mdl_cqq11,mdl_cqq3,mdl_cqq31,mdl_clq1,mdl_clq3,mdl_cee,mdl_cuu,mdl_cuu1,mdl_cdd,mdl_cdd1,mdl_ceu,mdl_ced,mdl_cud1,mdl_cud8,mdl_cle,mdl_clu,mdl_cld,mdl_cqe,
+            mdl_cqu1,mdl_cqu8,mdl_cqd1,mdl_cqd8,mdl_cledqre,mdl_cquqd1re,mdl_cquqd11re,mdl_cquqd8re,mdl_cquqd81re,mdl_clequ1re,mdl_clequ3re,mdl_cgtil,mdl_cwtil,mdl_chgtil,mdl_chwtil,mdl_chbtil,mdl_chwbtil,mdl_cewim,mdl_cebim,
+            mdl_cugim,mdl_cuwim,mdl_cubim,mdl_cdgim,mdl_cdwim,mdl_cdbim,mdl_chudim,mdl_cehim,mdl_cuhim,mdl_cdhim,mdl_cledqim,mdl_cquqd1im,mdl_cquqd8im,mdl_cquqd11im,mdl_cquqd81im,mdl_clequ1im,mdl_clequ3im,mdl_lambdasmeft,mdl_aew,
+            mdl_gf,as,mdl_linearpropcorrections,mdl_ymdo,mdl_ymup,mdl_yms,mdl_ymc,mdl_ymb,mdl_ymt,mdl_yme,mdl_ymm,mdl_ymtau;
+        }params_r_;
+
+        extern struct{
+            complex<double> mdl_g__exp__3,mdl_complexi,mdl_ceh,mdl_cuh,mdl_cdh,mdl_cew,mdl_ceb,mdl_cug,mdl_cuw,mdl_cub,mdl_cdg,mdl_cdw,mdl_cdb,mdl_chud,mdl_cledq,mdl_cquqd1,mdl_cquqd11,mdl_cquqd8,mdl_cquqd81,mdl_clequ1,mdl_clequ3,
+            mdl_ckm1x1,mdl_ckm1x2,mdl_ckm1x3,mdl_ckm2x1,mdl_ckm2x2,mdl_ckm2x3,mdl_ckm3x1,mdl_ckm3x2,mdl_ckm3x3,mdl_conjg__ckm1x1,mdl_conjg__ckm1x2,mdl_conjg__ckm1x3,mdl_conjg__ckm2x1,mdl_conjg__ckm2x2,mdl_conjg__ckm2x3,
+            mdl_conjg__ckm3x1,mdl_conjg__ckm3x2,mdl_conjg__ckm3x3;
+        }params_c_;
+
+        extern struct{
+            complex<double> gc_225, gc_229, gc_234, gc_238, gc_4, gc_136,
+            gc_197, gc_269, gc_270, gc_271, gc_272, gc_285, gc_319, gc_320,
+            gc_321, gc_322, gc_323, gc_324, gc_325, gc_326, gc_327,
+            gc_328, gc_329, gc_330, gc_352, gc_353, gc_364, gc_384, gc_387,
+            gc_392, gc_395, gc_398, gc_435, gc_436, gc_437, gc_438,
+            gc_464, gc_465, gc_466, gc_467, gc_708, gc_709, gc_718, gc_719,
+            gc_803, gc_804, gc_813, gc_814;
+        }couplings_; //you shouldn't have to edit these, but they are here for safekeeping!
+
+        void smatrixhel_(int pdgs[], int& procid, const int& npdg, double* p, double& alphas, double& scale2, int& nhel, double& ans);
+        void update_all_coup_();
+    }
+    void setDefaultMadgraphValues();
+    void initialize_madMELA();
+
+    extern TVar::VerbosityLevel myVerbosity_;
+    const pair<const complex<double>&, const complex<double>&> getCKMMatEl(int i, int j);
+
+    extern madMelaInput* madMelaCandidate;
```

Comment:
My understanding is that the biggest barrier to parallelization is the base FORTRAN - as the common blocks instantiate one space in memory that is always shared between subprocesses of the same compiled system. I actually have experimented with (and I think managed to get) some parallelization with MELA, but through a very complicated mess of compilation and a large headache.

Regardless, I see that there is the capability to have multiple MELACandidates - which is functionality I have never used personally. So yes, in that vein, I agree - I can try and interface with the MELACandidate system.

Reply:
> My understanding is that the biggest barrier to parallelization is the base FORTRAN - as the common blocks instantiate one space in memory that is always shared between subprocesses of the same compiled system. I actually have experimented with (and I think managed to get) some parallelization with MELA, but through a very complicated mess of compilation and a large headache.

Yes, we will have to sort those issues, but that will need some restructuring of the lower-level code. It would still go more smoothly if the functionality exposed to the user is more lenient - otherwise, we would be discussing rewriting MELA altogether.

> 
> Regardless, I see that there is the capability to have multiple MELACandidates - which is functionality I have never used personally. So yes, in that vein, I agree - I can try and interface with the MELACandidate system.

Reply:
Done.